### PR TITLE
Clarity functions typedoc annotations

### LINKED ISSUE #1245: [docs][transactions] add more detailed typedoc annotation to popular functions
for the @stacks/transactions package

- document most important annotations: `@param`, `@returns` for exported functions
- add [code blocks](https://typedoc.org/guides/doccomments/#code-blocks) with short examples (if it makes sense, use the same example for a test as well, especially if no tests exist yet)
- add ~`@internal`~ `@ignore` to functions that we only export to use ourselves (e.g. `omit`) and functions that might change function signature often

we can go through these by theme / "semantic categories" of functions; for example we can start with the functions "interacting with clarity values" (`uintCV`, `deserializeCV`, etc.); then continue with other important themes

### PR DESCRIPTION
**Pull request by ahsan-javaid**
## Description
- Add typedoc annotations for clarity functions
- Also add basic examples
- Refer to tests for more details

For details refer to issue #1245

## Type of Change
- [X] New feature
- [ ] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
No

## Testing information

Provide context on how tests should be performed.
`npm run test`

## Checklist
- [X] Code is commented where needed
- [ ] Unit test coverage for new or modified code paths
- [X] `npm run test` passes
- [ ] Changelog is updated
- [X] Tag reviewers

### CODE REVIEW

File: packages/transactions/src/clarity/types/intCV.ts [hunk 1]
```diff
@@ -31,6 +49,24 @@ interface UIntCV {
...
+ *
+ * @param {value} integer value to be converted to UIntCV clarity type (Only unsigned integer is allowed otherwise throws exception)
+ *
+ * @returns {UIntCV} returns instance of type UIntCV
+ *
+ * @example
+ * ```typescript
+ *  import { UIntCV } from '@stacks/transactions';
+ *
+ *  const value = UIntCV('100'); // parameter any of type: number | string | bigint | Uint8Array | BN
```
Comment (janniks) [inline packages/transactions/src/clarity/types/intCV.ts]: Should this be lowercase? Uppercase is the type, right?

File: packages/transactions/src/clarity/types/responseCV.ts [hunk 2]
```diff
@@ -13,10 +13,48 @@ interface ResponseOkCV<T extends ClarityValue = ClarityValue> {
...
+ *
+ * @param {value} ClarityValue value to be converted to responseErrorCV clarity type
+ *
+ * @returns {responseErrorCV} returns instance of type responseErrorCV
+ *
+ * @example
+ * ```typescript
+ *  import { responseErrorCV, intCV } from '@stacks/transactions';
+ *
+ *  const respOKCV = responseErrorCV(intCV(1));
```
Comment (janniks) [inline packages/transactions/src/clarity/types/responseCV.ts]: ok !== error

**Review by janniks — CHANGES_REQUESTED** [review]
Looks good! Minor comments,

File: packages/transactions/src/clarity/types/intCV.ts (hunk 1 above)
Comment (ahsan-javaid) [reply packages/transactions/src/clarity/types/intCV.ts]: Done!

File: packages/transactions/src/clarity/types/responseCV.ts (hunk 2 above)
Comment (ahsan-javaid) [reply packages/transactions/src/clarity/types/responseCV.ts]: Done!

**Pull request merged**